[Feature Request]  Update Python unbuffered I/O initialization for Python 3.12 and later

**Summary**

Compiling LAMMPS with Python 3.12 produces the following warning:
```
lammps/src/PYTHON/python_impl.cpp: In constructor ‘LAMMPS_NS::PythonImpl::PythonImpl(LAMMPS_NS::LAMMPS*)’:
lammps/src/PYTHON/python_impl.cpp:72:5: warning: ‘Py_UnbufferedStdioFlag’ is deprecated [-Wdeprecated-declarations]
   72 |     Py_UnbufferedStdioFlag = 1;
      |     ^~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/python3.12/Python.h:48,
                 from lammps/src/PYTHON/python_compat.h:17,
                 from lammps/src/PYTHON/python_impl.cpp:23:
/usr/include/python3.12/cpython/pydebug.h:20:37: note: declared here
   20 | Py_DEPRECATED(3.12) PyAPI_DATA(int) Py_UnbufferedStdioFlag;
      |                                     ^~~~~~~~~~~~~~~~~~~~~~
```

**Detailed Description**

Deprecated features will eventually go away and deprecation warnings confuse inexperienced LAMMPS users trying to compile LAMMPS without warnings or issues. Thus we will eventually need an update of the code.

**Further Information, Files, and Links**

https://docs.python.org/3/whatsnew/3.12.html#id6
https://docs.python.org/3/c-api/init.html

--- a/src/PYTHON/python_impl.cpp
+++ b/src/PYTHON/python_impl.cpp
@@ -61,16 +61,19 @@ PythonImpl::PythonImpl(LAMMPS *lmp) : Pointers(lmp)
   nfunc = 0;
   pfuncs = nullptr;
 
-#if PY_MAJOR_VERSION >= 3
-#ifndef Py_LIMITED_API
+#if PY_MAJOR_VERSION >= 3 && !defined(Py_LIMITED_API)
   // check for PYTHONUNBUFFERED environment variable
   const char *PYTHONUNBUFFERED = getenv("PYTHONUNBUFFERED");
+  // Force the stdout and stderr streams to be unbuffered.
+  bool unbuffered = PYTHONUNBUFFERED != nullptr && strcmp(PYTHONUNBUFFERED, "1") == 0;
 
-  if (PYTHONUNBUFFERED != nullptr && strcmp(PYTHONUNBUFFERED, "1") == 0) {
-    // Python Global configuration variable
-    // Force the stdout and stderr streams to be unbuffered.
-    Py_UnbufferedStdioFlag = 1;
-  }
+#if PY_VERSION_HEX >= 0x030800f0
+  PyConfig config;
+  PyConfig_InitPythonConfig(&config);
+  config.buffered_stdio = !unbuffered;
+#else
+  // Python Global configuration variable
+  Py_UnbufferedStdioFlag = unbuffered;
 #endif
 #endif
 
@@ -94,12 +97,17 @@ PythonImpl::PythonImpl(LAMMPS *lmp) : Pointers(lmp)
 #endif
 #endif
 
+#if PY_VERSION_HEX >= 0x030800f0 && !defined(Py_LIMITED_API)
+  Py_InitializeFromConfig(&config);
+  PyConfig_Clear(&config);
+#else
   Py_Initialize();
+#endif
 
   // only needed for Python 2.x and Python 3 < 3.7
   // With Python 3.7 this function is now called by Py_Initialize()
   // Deprecated since version 3.9, will be removed in version 3.11
-#if PY_MAJOR_VERSION < 3 || PY_MINOR_VERSION < 7
+#if PY_VERSION_HEX < 0x030700f0
   if (!PyEval_ThreadsInitialized()) { PyEval_InitThreads(); }
 #endif
 

